src/repeat.js

```diff
@@ -94,17 +115,74 @@ export class Repeat {
     }
   }
 
-  processItemsByStrategy() {
+  getInnerCollection() {
+    let expression = unwrapExpression(this.sourceExpression);
+    if (!expression) {
+      return null;
+    }
+    return expression.evaluate(this.scope, null);
+  }
+
+  observeInnerCollection() {
+    let items = this.getInnerCollection();
+    if (items instanceof Array) {
+      this.collectionObserver = this.observerLocator.getArrayObserver(items);
+    } else if (items instanceof Map) {
+      this.collectionObserver = this.observerLocator.getMapObserver(items);
+    } else {
+      return false;
+    }
+    this.callContext = 'handleInnerCollectionChanges';
+    this.collectionObserver.subscribe(this.callContext, this);
+    return true;
+  }
```

Comment:
That makes sense- the one thing we'd need to account for is the "inner collection" could be of a different type than the collection returned by the value converter.  

Contrived example: `repeat.for="myMap| toArray | sort:'key' | take:5"`.  The inner collection is a Map instance and the outer collection/strategy is an array.